rfcs/text/0030/file_description.yml

```diff
@@ -0,0 +1,271 @@
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+---
+- name: fd
+  title: File description
+  group: 2
+  short: These fields contain information about a file descriptor.
+  description: >
+    These fields contain information about a file descriptor.
+
+    In Unix and Unix-like computer operating systems, a file descriptor (FD, less frequently fildes) is a unique identifier (handle) for a file or other input/output resource, such as a pipe or network socket.
+  type: group
+  reusable:
+    top_level: false
+    expected:
+      - at: process
+        as: fds
+        short_override: An array of file descriptors for the process.
```

Comment:
My understanding of your approach here is that you want to re-use this field set at process as an array. Unfortunately our current tooling doesn't have a way of representing an array of an entire fieldset. 

Beyond our tooling, there's a larger question around how you want to represent arrays of objects. There's the regular object type, and then [nested](https://www.elastic.co/guide/en/elasticsearch/reference/current/nested.html), each of which have their own pros and cons when it comes to usage.

Reply:
ah yea, i will remove this and add it as a field to process.yml instead. @m-sample I know you wanted something a little more verbose like  process.fds.fd0 , but depending on how many fds we plan to capture, that structure might be a little too noisy and repetive. I don't imagine we will write too many search queries against fds, so maybe 'nested' type is ok to use, as it would be a special case search.

Reply:
If we capture file access telemetry events separately it's probably fine to search on those and leave this as an array of file descriptions with the reduced search-ability that implies if one avoid fan-out into multiple docs for the array entries.

It does bring up a future discussion point if we aggregate file accesses into single events/documents - do we use an array here & what are the likely search constraints and would an array be sufficient.

Reply:
So if I understand correctly, we can't have an array of a fieldset? Even if defined like so (in process.yml):
` 115     - name: fds
  116       level: extended
  117       type: file_description
  118       short: An array of the lowest file descriptors for the process.
  119       description: >
  120         An array of the lowest file descriptors for the process.
  121 
  122         Typically includes fd0(stdin), fd1(stdout), fd2(stderr), but can include other non character devices such as files, sockets, pipes and FIFOs.
  123 
  124       example: >
  125         [
  126           {
  127             descriptor:0,
  128             type: 'char_device',
  129             char_device: {
  130               major: 8,
  131               minor: 1
  132             }
  133           },
  134           {
  135             descriptor:1,
  136             type:'char_device',
  137             char_device: {
  138               major: 8,
  139               minor: 1
  140             }
  141           },
  142           {
  143             descriptor:9,
  144             type:'pipe',
  145             pipe: {
  146               inode: '6183207'
  147             }
  148           }
  149         ]
  150 
  151       normalize:
  152         - array
`

Reply:
@mitodrummer correct, our current tooling does not support it

Reply:
Ok i've updated process.fds to be of type 'object' as an array. That being said, i've left the new file_description fieldset as it is still nested under process.io.fd and process.tty. I'm wondering if it makes sense to mention that the objects in process.fds conform to the shape of type "file_description" in it's description or not. We do mention the fact that it is an array of descriptors, so maybe that is good enough..